File: quizz/Pérez/U2/Q05PairProgramming/Q05PairProgramming/ec/edu/espe/petfoodstore/utils/FileManager.js
import fs from 'fs';

export default class FileManager {
    static load() {
        try {
            if (fs.existsSync('food.json')) { // Verificar si el archivo existe
                const data = fs.readFileSync('food.json', 'utf8');
                const parsedData = JSON.parse(data);
                return Array.isArray(parsedData) ? parsedData : []; // Asegurarse de devolver un array
            } else {
                return [];
            }
        } catch (err) {
            console.error('Error reading file:', err);
            return [];
        }
    }

    static save(foodList) {
        fs.writeFile('food.json', JSON.stringify(foodList, null, 2), (err) => {
            if (err) {
                console.error('Error writing file:', err);
            } else {
                console.log('Food data saved to food.json');
            }
        });
    }
}

File: quizz/travez/u2/Q03PairProgramming/utils/file_manager.py
import json
import pickle

class FileManager:
    @staticmethod
    def load(file_name: str):
        with open(file_name, 'r') as f:
            return json.load(f)
        
    @staticmethod
    def save_obj(pc, file_path):
        try:
            with open(file_path, 'wb') as file:
                pickle.dump(pc, file)
            print(f"PC data saved to " + file_path)
        except IOError as e:
            print(f"Error saving file " + file_path)
